config/initializers/_patch.rb

```diff
@@ -17,4 +17,29 @@ def encode(string)
       data
     end
   end
+
+  # Dump Proxy pac to file system
+  class ProxyPacLoader
```

Comment:
I suggest to add some kind of an abstraction layer to your app. Maybe something like this:

`lib/proxy_pac_loader.rb`
~~~ruby
class ProxyPacLoader < ProxyPacRb::ProxyPacLoader
    def initialize
      @loaders = []
      @loaders << ProxyPacStringLoader.new
      @loaders << ProxyPacUriLoader.new
      @loaders << ProxyPacNullLoader.new

      @default_loader = -> { ProxyPacNullLoader.new }
  end
end
~~~


`lib/proxy_pac_parser.rb`
~~~ruby
class ProxyPacParser < ProxyPacRb::Parser
    def initialize(environment:)
      @parser = ProxyPacRb::ProxyPacParser.new(environment)
      @loader = ProxyPacLoader.new
      @linter = ProxyPacRb::ProxyPacLinter.new
    end
end
~~~